stars-core/src/test/kotlin/tools/aqua/stars/core/hooks/MinNodesInTSCHookTest.kt

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2024 The STARS Project Authors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package tools.aqua.stars.core.hooks
+
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import org.junit.jupiter.api.assertThrows
+import tools.aqua.stars.core.*
+import tools.aqua.stars.core.evaluation.TSCEvaluation
+import tools.aqua.stars.core.hooks.defaulthooks.MinNodesInTSCHook
+import tools.aqua.stars.core.metric.metrics.evaluation.SegmentCountMetric
+import tools.aqua.stars.core.tsc.builder.tsc
+
+/** MinNodesInTSCHookTest. */
+class MinNodesInTSCHookTest {
+  /** Test MinNodesInTSCHook with fail policy OK. */
+  @Test
+  fun `Test MinNodesInTSCHook with fail policy OK`() {
+    val tsc =
+        tsc<
+            SimpleEntity,
+            SimpleTickData,
+            SimpleSegment,
+            SimpleTickDataUnit,
+            SimpleTickDataDifference> {
+          any("") {}
+        }
+    TSCEvaluation(tsc = tsc, segments = segments()).apply {
+      val segmentCountMetric = setup()
+      registerPreEvaluationHooks(
+          MinNodesInTSCHook(minNodes = 2, failPolicy = EvaluationHookResult.OK))
+
+      runEvaluation(writePlots = false, writePlotDataCSV = false)
+
+      assertEquals(1, segmentCountMetric.getState())
+    }
+  }
+
+  /** Test MinNodesInTSCHook with fail policy SKIP. */
+  @Test
+  fun `Test MinNodesInTSCHook with fail policy SKIP`() {
+    val tsc =
+        tsc<
+            SimpleEntity,
+            SimpleTickData,
+            SimpleSegment,
+            SimpleTickDataUnit,
+            SimpleTickDataDifference> {
+          any("") {}
+        }
+    TSCEvaluation(tsc = tsc, segments = segments()).apply {
+      val segmentCountMetric = setup()
+      registerPreEvaluationHooks(
+          MinNodesInTSCHook(minNodes = 2, failPolicy = EvaluationHookResult.SKIP))
+
+      runEvaluation(writePlots = false, writePlotDataCSV = false)
+
+      assertEquals(0, segmentCountMetric.getState())
```

Comment:
I was thinking: Can we actually check that a specific EvaluationHookResult was achieved? Would it be interesting for us to have the runEvaluation() function to return a list of Map<Source, HookResult>?